src/activities/SetGlobalProperty.ts

```diff
@@ -0,0 +1,64 @@
+import type { IActivityHandler } from "@geocortex/workflow/runtime";
+import config from "@arcgis/core/config";
+
+interface SetGlobalPropertyInputs {
+    /**
+     * @description The name of the global property to modify.
+     * @required
+     * @noExpressions
+     */
+    name:
+        | "apiKey"
+        | "fontsUrl"
+        | "geoRSSServiceUrl"
+        | "geometryServiceUrl"
+        | "kmlServiceUrl"
+        | "routeServiceUrl";
+
+    /**
+     * @description The value to apply to the property.
+     * @required
+     */
+    value: string;
+}
+
+interface SetGlobalPropertyOutputs {}
+
+/**
+ * @category ArcGIS Maps SDK for JavaScript
+ * @clientOnly
+ * @description Configure global properties of the ArcGIS Maps SDK for JavaScript library.
+ * @helpUrl https://developers.arcgis.com/javascript/latest/api-reference/esri-config.html
+ * @supportedApps EXB, GWV
+ */
+export default class SetGlobalProperty implements IActivityHandler {
+    execute(inputs: SetGlobalPropertyInputs): SetGlobalPropertyOutputs {
+        const { name, value } = inputs;
+        if (!name) {
+            throw new Error("name is required");
+        }
+
+        // Ensure the name input only includes a subset of config properties
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        const prop = name as keyof Pick<
+            config,
+            SetGlobalPropertyInputs["name"]
+        >;
+
+        switch (prop) {
+            case "apiKey":
+            case "fontsUrl":
+            case "kmlServiceUrl":
+            case "geoRSSServiceUrl":
+            case "geometryServiceUrl":
+            case "routeServiceUrl":
+                config[name] = value;
+                break;
+            default:
+                prop satisfies never;
```

Comment:
I've not seen this before. What does it mean?

Reply:
`satisfies` is a type assertion. It is kind of like `as` but it is really just concerned with validation.

In this usage it ensures that the switch statement provides an exhaustive list of cases. We still require the `default` case to handle unexpected values at runtime, but it should never be hit according to the typings. 

You can see it in action if you add another legal property to the `name` input. The default case of the switch will now show a typing error.

Reply:
Nice. That's a handy mechanism. I'll need to use that in future.